GitUI/ScriptsEngine/ScriptOptionsParser.cs

```diff
@@ -106,6 +110,37 @@ public static (string? arguments, bool abort) Parse(string? arguments, IGitUICom
             List<IGitRef> currentBranches = new();
             List<IGitRef> currentTags = new();
 
+            string userInputCaption = string.Format(TranslatedStrings.ScriptUserInputCaption, scriptName);
+
+            // Specific handling of "UserInput" because the value entered should not replace all the variables
+            Match match;
+            while ((match = userInputRegex.Match(arguments)).Success)
+            {
+                Group defaultValueMatch = match.Groups["defaultValue"];
+                (string? arguments, bool abort) defaultValue = defaultValueMatch == null
+                    ? (string.Empty, false)
+                    : Parse(scriptName, defaultValueMatch.Value ?? string.Empty, uiCommands, owner, scriptOptionsProvider);
+
+                if (defaultValue.abort)
+                {
+                    return (arguments: null, abort: true);
+                }
+
+                string label = match.Groups["label"].Value;
+
+                using (IUserInputPrompt prompt = uiCommands.GetService<ISimplePromptCreator>().Create(userInputCaption, label, defaultValue.arguments))
+                {
+                    DialogResult result = prompt.ShowDialog(owner);
+                    if (result != DialogResult.OK)
+                    {
+                        throw new OperationCanceledException();
```

Comment:
😨 Why are we throwing here?
It looks as though we're using exceptions for the flow control, which is a code smell.
Throwing exceptions is quite expensive, and they represent an exceptional state, which user cancelling the dialog isn't.

Reply:
>Throwing exceptions is quite expensive

Performance doesn't matter here because the exception is thrown only one time to display a user popup.

>and they represent an exceptional state, which user cancelling the dialog isn't.

I think the cancelling of the dialog is exceptional otherwise we would have already an issue opened because the error popup is absolutely not good 🤣 

I really think a refactoring here is overkill as the method is returning at the moment only a value and we will have to refactor it to return also a state enum (Success/Abort/UserAbort), handle well this state for all arguments and handle it in the calling code. I predict more than 100 lines of new code and changes just to handle the case of 2 arguments `UserFile` and `UserInput` that ask inputs from user and that a bad/no input should abort the script. That's the handling of these 2 arguments and the possibility to abort which is "exceptional" 😉

I really don't want to go this way....